File: frontend/steam-gametime-frontend/src/app/models/profile.dto.ts
export interface ProfileDto {
    steamId64: string;
    personaName: string;
    steamLevel?: number;
    avatarSmall?: string;     // 32x32
    avatarMedium?: string;    // 64x64
    avatarFull?: string;      // 184x184
    countryCode?: string;     // e.g., "US"
    communityVisibilityState: number; // 1=Private, 3=Public
    personaState?: number;       // 0=Offline.. etc.
    timeCreatedUtc?: string;
    lastLogOffUtc?: string;
}

File: backend/Steam API/Dto/Output/ProfileDto.cs
namespace Steam_API.Dto.Output
{
    /// <summary>Minimal Steam profile information.</summary>
    public sealed class ProfileDto
    {
        public string SteamId64 { get; init; } = default!;
        public string PersonaName { get; init; } = "";
        public int? SteamLevel { get; init; }
        public string? AvatarSmall { get; init; }     // 32x32
        public string? AvatarMedium { get; init; }    // 64x64
        public string? AvatarFull { get; init; }      // 184x184
        public string? CountryCode { get; init; }     // e.g., "US"
        public int CommunityVisibilityState { get; init; } // 1=Private, 3=Public
        public int? PersonaState { get; init; }       // 0=Offline.. etc.
        public DateTimeOffset? TimeCreatedUtc { get; init; }
        public DateTimeOffset? LastLogOffUtc { get; init; }
    }
}


